dotnet/src/Planners/Planners.Core/Handlebars/HandlebarsTemplateEngineExtensions.cs

```diff
@@ -0,0 +1,393 @@
+                        }
+                    }
+                }
+                else
+                {
+                    // Process positional arguments
+                    var requiredParameters = functionView.Parameters.Where(p => p.IsRequired == true).ToList();
+                    if (arguments.Length >= requiredParameters.Count && arguments.Length <= functionView.Parameters.Count)
+                    {
+                        var argIndex = 0;
+                        foreach (var arg in arguments)
+                        {
+                            var param = functionView.Parameters[argIndex];
+                            if (IsExpectedParameterType(param.Type.ToString() ?? "", arg.GetType().Name, arg))
+                            {
+                                variables[param.Name] = arguments[argIndex];
+                                argIndex++;
+                            }
+                            else
+                            {
+                                throw new SKException($"Invalid parameter type for function {functionView.Name}. Parameter {param.Name} expects type {param.Type} but received {arguments[argIndex].GetType()}.");
+                            }
+                        }
+                    }
+                    else
+                    {
+                        throw new SKException($"Invalid parameter count for function {functionView.Name}. {arguments.Length} were specified but {functionView.Parameters.Count} are required.");
+                    }
+                }
+            }
+
+            foreach (var v in variables)
+            {
+                var varString = v.Value?.ToString() ?? "";
+                if (executionContext.Variables.TryGetValue(v.Key, out var argVal))
+                {
+                    executionContext.Variables[v.Key] = varString;
+                }
+                else
+                {
+                    executionContext.Variables.Add(v.Key, varString);
+                }
+            }
+
+            // TODO (@teresaqhoang): Add model results to execution context + test possible deadlock scenario
+            ISKFunction function = kernel.Functions.GetFunction(functionView.PluginName, functionView.Name);
+
+#pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
+            KernelResult result = kernel.RunAsync(executionContext.Variables, cancellationToken, function).GetAwaiter().GetResult();
+#pragma warning restore VSTHRD002 // Avoid problematic synchronous waits
+
+            // Write the result to the template
+            return result.GetValue<object?>();
+        });
+    }
+
+    private static void RegisterSystemHelpers(
+        IHandlebars handlebarsInstance,
+        Dictionary<string, object?> variables
+    )
+    {
+        handlebarsInstance.RegisterHelper("array", (in HelperOptions options, in Context context, in Arguments arguments) =>
```

Comment:
Question only, are these covered by the standard Handlebars helpers?

From https://github.com/Handlebars-Net/Handlebars.Net/tree/master/source/Handlebars/Helpers

Reply:
Ah haven't checked those yet, I'll look through those and remove any duped functionality